soilmask.py

```diff
@@ -0,0 +1,206 @@
+    """
+    rgb_mask = cv2.bitwise_and(img, img, mask=bin_mask)
+
+    return rgb_mask
+
+
+def gen_plant_mask(color_img: np.ndarray, kernel_size: int = 3) -> np.ndarray:
+    """Generates an image with plants masked in.
+    Arguments:
+        color_img: RGB image to mask
+        kernel_size: masking kernel size
+    Return:
+        An RGB image with plants masked in
+    """
+    r_channel = color_img[:, :, 2]
+    g_channel = color_img[:, :, 1]
+    b_channel = color_img[:, :, 0]
+
+    sub_img = (g_channel.astype('int') - r_channel.astype(
+        'int') - 0) > 0  # normal: -2
+
+    mask = np.zeros_like(b_channel)
+
+    mask[sub_img] = MAX_PIXEL_VAL
+
+    blur = cv2.blur(mask, (kernel_size, kernel_size))
+    pix = np.array(blur)
+    sub_mask = pix > 128
+
+    mask_1 = np.zeros_like(b_channel)
+    mask_1[sub_mask] = MAX_PIXEL_VAL
+
+    return mask_1
+
+def get_epsg(filename):
+    """Returns the EPSG of the georeferenced image file
+    Args:
+        filename(str): path of the file to retrieve the EPSG code from
+    Return:
+        Returns the found EPSG code, or None if it's not found or an error ocurred
+    """
+    logger = logging.getLogger(__name__)
+
+    try:
+        src = gdal.Open(filename)
+
+        proj = osr.SpatialReference(wkt=src.GetProjection())
+
+        return proj.GetAttrValue('AUTHORITY', 1)
+    # pylint: disable=broad-except
+    except Exception as ex:
+        logger.warn("[get_epsg] Exception caught: %s", str(ex))
+    # pylint: enable=broad-except
+
+    return None
+
+
+def get_image_file_geobounds(filename):
+    """Uses gdal functionality to retrieve rectilinear boundaries from the file
+    Args:
+        filename(str): path of the file to get the boundaries from
+    Returns:
+        The upper-left and calculated lower-right boundaries of the image in a list upon success.
+        The values are returned in following order: min_y, max_y, min_x, max_x. A list of numpy.nan
+        is returned if the boundaries can't be determined
+    """
+    try:
+        str_filename = str(filename)
+        epsg = get_epsg(str_filename)
+        if str(epsg) != "4326":
+            _, tmp_name = tempfile.mkstemp()
+            src = gdal.Open(str_filename)
+            gdal.Warp(tmp_name, src, dstSRS='EPSG:4326')
+            str_filename = tmp_name
+
+        src = gdal.Open(str_filename)
+        ulx, xres, _, uly, _, yres = src.GetGeoTransform()
+        lrx = ulx + (src.RasterXSize * xres)
+        lry = uly + (src.RasterYSize * yres)
+
+        min_y = min(uly, lry)
+        max_y = max(uly, lry)
+        min_x = min(ulx, lrx)
+        max_x = max(ulx, lrx)
+
+        return [min_y, max_y, min_x, max_x]
+    except Exception as ex:
+        logging.info("[image_get_geobounds] Exception caught: %s", str(ex))
+
+    return [np.nan, np.nan, np.nan, np.nan]
+
+
+def create_geotiff(pixels, gps_bounds, out_path, nodata=-99, asfloat=False, extractor_info=None, system_md=None,
```

Comment:
Also missing type annotations, and these parameters are not used at all:

- `extractor_info`
- `system_md`
- `extra_metadata`